### PR TITLE
Skip redundant superclasses\supertraits.

### LINKED ISSUE #1209: Skip redundant superclasses\supertraits.
https://github.com/scala/scala-dev/issues/98#issuecomment-194603549

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

**Comment by nicolasstucki** [comment]
Review by @DarkDimius

File: src/dotty/tools/backend/jvm/CollectSuperCalls.scala [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package dotty.tools.backend.jvm
+
+import dotty.tools.dotc.ast.tpd._
+import dotty.tools.dotc.ast.Trees
+import dotty.tools.dotc.core.Contexts.Context
+import dotty.tools.dotc.core.Symbols._
+import dotty.tools.dotc.transform.TreeTransforms.{MiniPhaseTransform, TransformerInfo}
+
+class CollectSuperCalls extends MiniPhaseTransform {
```
Comment (smarter) [inline src/dotty/tools/backend/jvm/CollectSuperCalls.scala]: Please add a comment explaining what the purpose of this phase is in some details

Comment (nicolasstucki) [inline src/dotty/tools/backend/jvm/CollectSuperCalls.scala]: I knew that I was missing something important but could not remember what.

**Comment by DarkDimius** [comment]
LGTM

**Pull request merged**